pkg/kubectl/cmd/set/set_data.go

```diff
@@ -0,0 +1,377 @@
+		if patch.Err != nil {
+			allErrs = append(allErrs, fmt.Errorf("error: %s/%s %v\n", info.Mapping.Resource, info.Name, patch.Err))
+			continue
+		}
+
+		//no changes
+		if string(patch.Patch) == "{}" || len(patch.Patch) == 0 {
+			allErrs = append(allErrs, fmt.Errorf("info: %s %q was not changed\n", info.Mapping.Resource, info.Name))
+			continue
+		}
+
+		if o.PrintObject != nil && (o.Local || o.DryRun) {
+			if err := o.PrintObject(o.Cmd, o.Local, o.Mapper, patch.Info.AsVersioned(), o.Out); err != nil {
+				return err
+			}
+			continue
+		}
+
+		obj, err := resource.NewHelper(info.Client, info.Mapping).Patch(info.Namespace, info.Name, types.StrategicMergePatchType, patch.Patch)
+		if err != nil {
+			allErrs = append(allErrs, fmt.Errorf("failed to patch data to : %v\n", err))
+			continue
+		}
+		info.Refresh(obj, true)
+
+		if len(o.Output) > 0 {
+			if err := o.PrintObject(o.Cmd, o.Local, o.Mapper, info.AsVersioned(), o.Out); err != nil {
+				return err
+			}
+			continue
+		}
+		o.PrintSuccess(o.Mapper, o.ShortOutput, o.Out, info.Mapping.Resource, info.Name, o.DryRun, "data updated")
+	}
+	return utilerrors.NewAggregate(allErrs)
+}
+
+// StructuredGenerate outputs a secret object using the configured fields
+func StructuredGenerate(fileSources, literalSources []string, envFileSource string) (map[string][]byte, error) {
+
+	data := map[string][]byte{}
+	if len(fileSources) > 0 {
+		if err := handleFromFileSources(data, fileSources); err != nil {
+			return nil, err
+		}
+	}
+	if len(literalSources) > 0 {
+		if err := handleFromLiteralSources(data, literalSources); err != nil {
+			return nil, err
+		}
+	}
+	if len(envFileSource) > 0 {
+		if err := handleFromEnvFileSource(data, envFileSource); err != nil {
+			return nil, err
+		}
+	}
+	
+	return data, nil
+}
+
+//Note: the obj mutates in the function
+func updateDataForObject(obj runtime.Object, data map[string][]byte, fn func(map[string][]byte, map[string][]byte) (bool, map[string][]byte)) (bool, error) {
+	switch t := obj.(type) {
+	case *core.Secret:
+		transformed, result := fn(t.Data, data)
+		t.Data = result
+		return transformed, nil
+	/*case *v1.ConfigMap:
+		transformed, result := fn(t.Data, data)
+		t.Data = result
+		return transformed, nil*/
+	default:
+		return false, fmt.Errorf("setting data is only supported for Secrets and ConfigMaps")
+	}
+}
+
+func addData(existing map[string][]byte, target map[string][]byte) (bool, map[string][]byte) {
+	transformed := false
+	updated := existing
+	for k, v := range target {
+		if !containData(existing, k) {
+			existing[k] = v
+			transformed = true
+		}
+	}
+	return transformed, updated
+}
+
+func containData(slice map[string][]byte, item string) bool {
+	for k := range slice {
+		if k == item {
+			return true
+		}
+	}
+	return false
+}
+
+// handleFromLiteralSources adds the specified literal source information into the provided secret
+func handleFromLiteralSources(data map[string][]byte, literalSources []string) error {
```

Comment:
I'm pretty confident that all this code is copy/pasted from somewhere else. Can we not do that?

Reply:
I have seen these functions somewhere else. I know you've copied them from somewhere else. Please don't.